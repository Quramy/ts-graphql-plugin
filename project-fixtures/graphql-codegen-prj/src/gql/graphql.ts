/* eslint-disable */
import { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';
export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
export type MakeEmpty<T extends { [key: string]: unknown }, K extends keyof T> = { [_ in K]?: never };
export type Incremental<T> = T | { [P in keyof T]?: P extends ' $fragmentName' | '__typename' ? T[P] : never };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: { input: string; output: string; }
  String: { input: string; output: string; }
  Boolean: { input: boolean; output: boolean; }
  Int: { input: number; output: number; }
  Float: { input: number; output: number; }
};

export type Post = {
  __typename?: 'Post';
  author: User;
  body: Scalars['String']['output'];
  createdAt: Scalars['String']['output'];
  id: Scalars['ID']['output'];
  title: Scalars['String']['output'];
  updatedAt: Scalars['String']['output'];
};

export type Query = {
  __typename?: 'Query';
  popularPosts: Array<Post>;
};

export type User = {
  __typename?: 'User';
  avatarURL: Scalars['String']['output'];
  createdAt: Scalars['String']['output'];
  id: Scalars['ID']['output'];
  name: Scalars['String']['output'];
  updatedAt: Scalars['String']['output'];
};

export type PopularPosts_QueryQueryVariables = Exact<{ [key: string]: never; }>;


export type PopularPosts_QueryQuery = { __typename?: 'Query', popularPosts: Array<(
    { __typename?: 'Post', id: string }
    & { ' $fragmentRefs'?: { 'PostSummary_PostFragment': PostSummary_PostFragment } }
  )> };

export type PostSummary_PostFragment = { __typename?: 'Post', id: string, title: string, author: (
    { __typename?: 'User', name: string }
    & { ' $fragmentRefs'?: { 'UserAvatar_UserFragment': UserAvatar_UserFragment } }
  ) } & { ' $fragmentName'?: 'PostSummary_PostFragment' };

export type UserAvatar_UserFragment = { __typename?: 'User', name: string, avatarURL: string } & { ' $fragmentName'?: 'UserAvatar_UserFragment' };

export const UserAvatar_UserFragmentDoc = {"kind":"Document","definitions":[{"kind":"FragmentDefinition","name":{"kind":"Name","value":"UserAvatar_User"},"typeCondition":{"kind":"NamedType","name":{"kind":"Name","value":"User"}},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"name"}},{"kind":"Field","name":{"kind":"Name","value":"avatarURL"}}]}}]} as unknown as DocumentNode<UserAvatar_UserFragment, unknown>;
export const PostSummary_PostFragmentDoc = {"kind":"Document","definitions":[{"kind":"FragmentDefinition","name":{"kind":"Name","value":"PostSummary_Post"},"typeCondition":{"kind":"NamedType","name":{"kind":"Name","value":"Post"}},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"id"}},{"kind":"Field","name":{"kind":"Name","value":"title"}},{"kind":"Field","name":{"kind":"Name","value":"author"},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"name"}},{"kind":"FragmentSpread","name":{"kind":"Name","value":"UserAvatar_User"}}]}}]}},{"kind":"FragmentDefinition","name":{"kind":"Name","value":"UserAvatar_User"},"typeCondition":{"kind":"NamedType","name":{"kind":"Name","value":"User"}},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"name"}},{"kind":"Field","name":{"kind":"Name","value":"avatarURL"}}]}}]} as unknown as DocumentNode<PostSummary_PostFragment, unknown>;
export const PopularPosts_QueryDocument = {"kind":"Document","definitions":[{"kind":"OperationDefinition","operation":"query","name":{"kind":"Name","value":"PopularPosts_Query"},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"popularPosts"},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"id"}},{"kind":"FragmentSpread","name":{"kind":"Name","value":"PostSummary_Post"}}]}}]}},{"kind":"FragmentDefinition","name":{"kind":"Name","value":"UserAvatar_User"},"typeCondition":{"kind":"NamedType","name":{"kind":"Name","value":"User"}},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"name"}},{"kind":"Field","name":{"kind":"Name","value":"avatarURL"}}]}},{"kind":"FragmentDefinition","name":{"kind":"Name","value":"PostSummary_Post"},"typeCondition":{"kind":"NamedType","name":{"kind":"Name","value":"Post"}},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"id"}},{"kind":"Field","name":{"kind":"Name","value":"title"}},{"kind":"Field","name":{"kind":"Name","value":"author"},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"name"}},{"kind":"FragmentSpread","name":{"kind":"Name","value":"UserAvatar_User"}}]}}]}}]} as unknown as DocumentNode<PopularPosts_QueryQuery, PopularPosts_QueryQueryVariables>;