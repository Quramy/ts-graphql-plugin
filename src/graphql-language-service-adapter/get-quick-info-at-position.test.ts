import ts from 'typescript';
import { mark, Frets } from 'fretted-strings';
import { createSimpleSchema } from './testing/simple-schema';
import { AdapterFixture } from './testing/adapter-fixture';
import { GraphQLSchema } from 'graphql';

function delegateFn(): ts.QuickInfo {
  return {
    kind: ts.ScriptElementKind.string,
    kindModifiers: '',
    textSpan: {
      start: 0,
      length: 0,
    },
    displayParts: [],
  };
}

function createFixture(name: string, schema?: GraphQLSchema) {
  return new AdapterFixture(name, schema);
}
describe('getQuickInfoAtPosition', () => {
  it('should return GraphQL quick info', () => {
    const fixture = createFixture('main.ts', createSimpleSchema());
    const quickInfoFn = fixture.adapter.getQuickInfoAtPosition.bind(fixture.adapter, delegateFn, 'main.ts');
    const frets: Frets = {};
    fixture.source = mark(
      `
        const query = \`
          query {
            hello
    %%%     ^   ^       %%%
    %%%     a1  a2      %%%
          }
        \`;
    `,
      frets,
    );
    expect(quickInfoFn(frets.a1.pos - 1)!.displayParts).toEqual([]);
    expect(
      quickInfoFn(frets.a1.pos)!
        .displayParts!.map(dp => dp.text)
        .join(''),
    ).toMatchSnapshot();
    expect(
      quickInfoFn(frets.a2.pos)!
        .displayParts!.map(dp => dp.text)
        .join(''),
    ).toMatchSnapshot();
    expect(quickInfoFn(frets.a2.pos + 1)!.displayParts).toEqual([]);
  });
});
