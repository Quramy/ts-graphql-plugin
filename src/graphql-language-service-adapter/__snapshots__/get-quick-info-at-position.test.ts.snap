// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getQuickInfoAtPosition should return GraphQL quick info 1`] = `"Query.hello: String!"`;

exports[`getQuickInfoAtPosition should return GraphQL quick info 2`] = `"Query.hello: String!"`;
